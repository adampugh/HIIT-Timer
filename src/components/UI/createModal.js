import React, { Component } from "react";
import { connect } from "react-redux";
import Modal from "react-modal";

import * as actions from "../../store/actions/actions";

const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      }
}

export class CreateModal extends Component { 
    state = {
        workoutUrl: ""
    }

    updateUrl = (e) => {
        this.setState({
            workoutUrl: e.target.value
        });
    }

    handleOnClick = () => {
        let workout = {
            title: this.state.workoutUrl,
            totalTime: 0,
            exercises: [],
            index: this.props.workoutIndex
        };
        this.props.startAddWorkout(workout);
    }

    render() {

        return (
            <Modal 
                isOpen={this.props.isOpen}
                contentLabel="create workout modal"
                appElement={document.getElementById('root')} 
                onRequestClose={this.props.handleCloseModal}
                style={customStyles}
                closeTimeoutMS={200}
                className="modal"
            >
            <div className="modal__box">
                <div className="modal__header">
                    <div></div>
                    <div><h3>Create Workout</h3></div>
                    <div className="modal__close">
                        <h3>
                            <span onClick={this.props.handleCloseModal}>
                                <i className="fas fa-times"></i>
                            </span>
                        </h3>
                    </div>
                </div>
                <hr />
                <div className="modal--create">
                    <label>Name</label>
                    <input type="text" maxLength="20" onChange={(e) => this.updateUrl(e)}/>
                    
                        <button 
                            onClick={this.handleOnClick} 
                            disabled={this.state.workoutUrl.length < 1} 
                            className="modal__button">
                            Create Workout
                        </button>
                   
                </div>
            </div>
        </Modal>
        )
    }
}    

const mapDispatchToProps = dispatch => {
    return {
        startAddWorkout: (workout) => dispatch(actions.startAddWorkout(workout))
    }
}


export default connect(null, mapDispatchToProps)(CreateModal);