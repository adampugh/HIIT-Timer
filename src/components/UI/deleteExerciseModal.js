import React, { Component } from "react";
import Modal from "react-modal";
import { connect } from "react-redux";

import * as actions from "../../store/actions/actions";

const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      },
      content : {
      }
}

// pass in index and dispatch action
class deleteExerciseModal extends Component {
    handleDelete(index, workoutId) {
        this.props.deleteExercise(index, workoutId);
        this.props.handleCloseModal();
    }


    render() {
        return (
            <Modal 
                isOpen={this.props.isOpen}
                contentLabel="login modal"
                appElement={document.getElementById('root')} 
                onRequestClose={this.props.handleCloseModal}
                style={customStyles}
                closeTimeoutMS={200}
                className="modal"
            >
                <div className="modal__box">
                    <div className="modal__header">
                        <div></div>
                        <div><h3>Delete {this.props.modalTitle}</h3></div>
                        <div className="modal__close">
                            <h3>
                                <span onClick={this.props.handleCloseModal}>
                                    <i className="fas fa-times"></i>
                                </span>
                            </h3>
                        </div>
                    </div>
                    <hr />
                    <p>Do you want to delete this {this.props.modalTitle.toLowerCase()}?</p>
                    <button className="modal__login-button" onClick={() => this.handleDelete(this.props.exerciseIndex, this.props.workoutId)} >
                        <span>Yes</span>
                    </button>
                    <button className="modal__login-button" onClick={this.props.handleCloseModal}>
                        <span>No</span>
                    </button>
                </div>
            </Modal>
        );
    }
};

const mapDispatchToProps = dispatch => {
    return {
        deleteExercise: (index, id) => dispatch(actions.deleteExercise(index, id))
    }
}

export default connect(null, mapDispatchToProps)(deleteExerciseModal);