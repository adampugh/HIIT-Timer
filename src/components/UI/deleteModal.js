import React from "react";
import Modal from "react-modal";



const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      },
      content : {
      }
}

const loginModal = (props) => (
    <Modal 
        isOpen={props.isOpen}
        contentLabel="login modal"
        appElement={document.getElementById('root')} 
        onRequestClose={props.handleCloseModal}
        style={customStyles}
        closeTimeoutMS={200}
        className="modal"
    >
        <div className="modal__box">
            <div className="modal__header">
                <div></div>
                <div><h3>Delete {props.modalTitle}</h3></div>
                <div className="modal__close">
                    <h3>
                        <span onClick={props.handleCloseModal}>
                            <i className="fas fa-times"></i>
                        </span>
                    </h3>
                </div>
            </div>
            <hr />
            <p>Do you want to delete this {props.modalTitle.toLowerCase()}?</p>
            <button className="modal__login-button">

                <span>Yes</span>
            </button>
            <button className="modal__login-button">

                <span>No</span>
            </button>
        </div>
    </Modal>
);

export default loginModal;