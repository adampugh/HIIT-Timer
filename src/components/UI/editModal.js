import React, { Component } from "react";
import Modal from "react-modal";
import { withFormik, Field, Form } from "formik";
import Yup from "yup";


const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      }
};

const editModal = (props) => (
    <Modal 
        isOpen={props.isOpen}
        contentLabel="add exercise modal"
        appElement={document.getElementById('root')} 
        onRequestClose={props.handleCloseModal}
        style={customStyles}
        closeTimeoutMS={200}
        className="modal modal--edit"
    >
        <div className="modal__box">
            <div className="modal__header">
                <div></div>
                <div><h3>Add Exercise</h3></div>
                <div className="modal__close">
                    <h3>
                        <span onClick={props.handleCloseModal}>
                            <i className="fas fa-times"></i>
                        </span>
                    </h3>
                </div>
            </div>
            <hr />
            <div className="modal--edit">
                <Form>
                    <label>Name</label>
                    <Field type="text" maxLength="20" name="exerciseName" />
                    <div className="exerciseGrid">
                        <div className="exerciseGrid__timeCol">
                            <label>Time</label>
                            <Field className="modal--edit__numberInput1" type="number" max="60" name="exerciseMinutes" />
                            :
                            <Field className="modal--edit__numberInput2" type="number" max="59" name="exerciseSeconds" />
                        </div>
                        <div className="exerciseGrid__colorCol">
                            <label>Color</label>
                            <Field component="select" name="exerciseColor" style={{backgroundColor: props.values.exerciseColor}}>
                                <option value="#40b3e1"></option>
                                <option value="#acff82"></option>
                                <option value="#fedc45"></option>
                                <option value="#fec345"></option>
                                <option value="#fe6145"></option>
                                <option value="#b145fe"></option>
                            </Field>
                        </div>
                    </div>
                    
                        <div>
                            <label className="modal--edit__breakLabel">Break</label>
                        </div>
                        <div className="modal--edit__checkboxDiv">
                            <Field id="switch" className="modal--edit__checkbox" type="checkbox" name="breakIncluded" checked={props.values.breakIncluded} />
                            <label for="switch">Toggle</label>
                        </div>
                    
                    { props.values.breakIncluded &&
                        <div className="exerciseGrid">
                            <div className="exerciseGrid__timeCol">
                                <label>Time</label>
                                <Field className="modal--edit__numberInput1" type="number" max="60" name="breakMinutes" />
                                :
                                <Field className="modal--edit__numberInput2" type="number" max="59" name="breakSeconds" />
                            </div>
                            <div className="exerciseGrid__colorCol">
                                <label>Color</label>
                                <Field component="select" name="breakColor" style={{backgroundColor: props.values.breakColor}}>
                                    <option value="#40b3e1"></option>
                                    <option value="#acff82"></option>
                                    <option value="#fedc45"></option>
                                    <option value="#fec345"></option>
                                    <option value="#fe6145"></option>
                                    <option value="#b145fe"></option>
                                </Field>
                            </div>
                        </div>
                    }
                    <button className="modal__button">Add Exercise</button>
                </Form>
            </div>
        </div>
    </Modal>
);

const formikEditModal = withFormik({
    mapPropsToValues() {
        return {
            exerciseName: "",
            exerciseMinutes: 0,
            exerciseSeconds: 0,
            exerciseColor: "",
            breakIncluded: true,
            breakName: "Break",
            breakMinutes: 0,
            breakSeconds: 0,
            breakColor: "grey"
        }
    },
    handleSubmit(values) {
        console.log(values);
    }
})(editModal);

export default formikEditModal;
// <div className="modal--edit__breakInclude">
// <div>
//     <label className="modal--edit__breakLabel">Break</label>
// </div>
// <div className="modal--edit__checkboxDiv">
//     <Field id="switch" className="modal--edit__checkbox" type="checkbox" name="breakIncluded" checked={props.values.breakIncluded} />
//     <label for="switch">Toggle</label>
// </div>
// </div>