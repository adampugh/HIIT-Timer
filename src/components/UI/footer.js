import React from "react";

import SocialLinks from "./socialLinks";

const footer = () => (
    <footer>
        <div className="container">
            <h1>HIIT Timer</h1>
            <SocialLinks />
            <h3>&copy; 2018 Design by <a href="https://adampugh.github.io/kokonoka/" target="_blank" rel="noopener noreferrer">Kokonoka </a></h3>
        </div>
    </footer>
);

export default footer;