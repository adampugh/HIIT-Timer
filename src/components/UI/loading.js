import React from "react";

import dumbbellIcon from "../../assets/photos/dumbbell-icon.png";


const loading = () => (
    <div className="loading">
        <img className="workoutsBanner__image" src={dumbbellIcon} alt="arm icon" />
        <h2>. . . Loading</h2>
    </div>
);

export default loading;