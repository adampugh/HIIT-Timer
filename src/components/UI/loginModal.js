import React from "react";
import Modal from "react-modal";

import GoogleIcon from "../../assets/photos/googlelogo.jpg";
import FacebookIcon from "../../assets/photos/facebooklogo.jpg";

const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      },
      content : {
      }
}

const loginModal = (props) => (
    <Modal 
        isOpen={props.isOpen}
        contentLabel="login modal"
        appElement={document.getElementById('root')} 
        onRequestClose={props.handleCloseModal}
        style={customStyles}
        closeTimeoutMS={200}
        className="modal"
    >
        <div className="modal__box">
            <div className="modal__header">
                <div></div>
                <div><h3>Log In</h3></div>
                <div className="modal__close">
                    <h3>
                        <span onClick={props.handleCloseModal}>
                            <i className="fas fa-times"></i>
                        </span>
                    </h3>
                </div>
            </div>
            <hr />

            <button className="modal__login-button">
                <img src={GoogleIcon} alt="google logo" />
                <span>Sign in with Google</span>
            </button>
            <button className="modal__login-button">
                <img src={FacebookIcon} alt="facebook logo" />
                <span>Sign in with Facebook</span>
            </button>
        </div>
    </Modal>
);

export default loginModal;