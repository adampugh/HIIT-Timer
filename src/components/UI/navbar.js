import React from "react";
import { NavLink } from "react-router-dom";

const navbar = (props) => (
    <nav className="navbar">
        <div className="container">
            <div className="navbar__items">
                <h2>HIIT Timer</h2>
                <div className="navbar__links">
                    {props.links.map(item => 
                        <NavLink 
                            key={item}
                            to={item.split(" ").join("").toLowerCase()}
                            activeClassName={"navbar__links--active"}>
                                {item}
                        </NavLink>
                    )}
                </div>
            </div>
        </div>
    </nav>
);

export default navbar;