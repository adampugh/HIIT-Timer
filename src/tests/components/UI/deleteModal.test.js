import React from "react";
import { shallow } from "enzyme"; 

import { DeleteModal } from "../../../components/UI/deleteModal";
import workoutState from "../../fixtures/workouts";

test("should render DeleteModal correctly", () => {
    const wrapper = shallow(<DeleteModal />);
    expect(wrapper).toMatchSnapshot();
});

test("should call startDeleteWorkout action in DeleteModal", () => {
    const workout = workoutState.workouts[0];
    const startDeleteWorkoutSpy = jest.fn();
    const wrapper = shallow(<DeleteModal 
        handleCloseModal={() => {}} 
        selectedWorkoutId={workout.id} 
        startDeleteWorkout={startDeleteWorkoutSpy} />);
    wrapper.find("button").first().simulate("click");
    expect(startDeleteWorkoutSpy).toHaveBeenCalledWith(workout.id);
});