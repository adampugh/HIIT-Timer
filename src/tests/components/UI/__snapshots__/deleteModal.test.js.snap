// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DeleteModal correctly 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={200}
  contentLabel="login modal"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "content": Object {},
      "overlay": Object {
        "background": "rgba(0, 0, 0, 0.6)",
        "zIndex": 2000,
      },
    }
  }
>
  <div
    className="modal__box"
  >
    <div
      className="modal__header"
    >
      <div />
      <div>
        <h3>
          Delete Workout
        </h3>
      </div>
      <div
        className="modal__close"
      >
        <h3>
          <span>
            <i
              className="fas fa-times"
            />
          </span>
        </h3>
      </div>
    </div>
    <hr />
    <p>
      Do you want to delete this workout?
    </p>
    <button
      className="modal__login-button"
      onClick={[Function]}
    >
      <span>
        Yes
      </span>
    </button>
    <button
      className="modal__login-button"
    >
      <span>
        No
      </span>
    </button>
  </div>
</Modal>
`;
